### ISSUE TITLE
Modal Bottom Sheet disappears when content changes size quickly

### ISSUE THREAD

**Issue by jacobras** [composablehorizons/composables-core#18]
I have a modal bottom sheet that shows content that could change. If the content changes and causes the size to change too quickly after showing the bottom sheet (which happens because data comes in asynchronously), the bottom sheet dismisses itself.

### Reproduction code

Here's a sample reproducing the issue by just varying the length of a text displayed. Using Compose BOM 2024.08.00 and Composables Core 1.11.1:

```kotlin
@Composable
fun ProblematicBottomSheet() {
    var linesOfText by remember { mutableIntStateOf(10) }

    LaunchedEffect(Unit) {
        delay(500)
        linesOfText = 5
    }

    val state = rememberModalBottomSheetState(initialDetent = SheetDetent.FullyExpanded)
    ModalBottomSheet(
        state = state
    ) {
        Scrim()
        Sheet(
            modifier = Modifier.background(Color.White)
        ) {
            Column(modifier = Modifier.fillMaxWidth()) {
                DragIndication(modifier = Modifier.padding(top = 22.dp))

                Text(
                    buildString {
                        repeat(linesOfText) {
                            appendLine("$linesOfText lines of text")
                        }
                    }
                )
            }
        }
    }
}
```

### Expected result

If I set the `delay` to 1500ms, the text just shortens and the bottom sheet stays open:

![ComposablesCoreBottomSheetExpected](https://github.com/user-attachments/assets/6e22dbfa-de55-4069-834b-66ab00d385fe)

### Actual result

but if I set the `delay` to 500ms, the text shortens quickly and it triggers dismissal of the bottom sheet:

![ComposablesCoreBottomSheetActual](https://github.com/user-attachments/assets/1c75220e-2587-4ea9-b459-65f9fe6ff1d1)

Here's the code above in a runnable project: [ComposablesBottomSheetReproduction.zip](https://github.com/user-attachments/files/16967567/ComposablesBottomSheetReproduction.zip)

**Comment by alexstyl** [comment]
Thank you for the detailed report. Will have a look later today

**Comment by jacobras** [comment]
Ah, yeah, I'm glad I mentioned API 35 in the Slack channel, forgot to write it in the bug report here.

> Seems like it is not a API issue but a race condition. This is more clear if you change the delay to something as short as delay(250)

That's good news imo, less stressful to fix then dealing with API issues ;-)

Note that I **personally** am not running this in production yet, so I have no rush with this issue.

**Comment by alexstyl** [comment]
Fixed in [1.11.2](https://github.com/composablehorizons/composables-core/releases/tag/1.11.2)
